smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/providers/connectors/ExecutionHolder.java

```diff
@@ -40,13 +46,26 @@ public ExecutionHolder(Vertx vertx) {
     }
 
     @Inject
-    public ExecutionHolder(Instance<Vertx> instanceOfVertx) {
-        if (instanceOfVertx == null || instanceOfVertx.isUnsatisfied()) {
+    public ExecutionHolder(Instance<Vertx> instanceOfVertx, Instance<Config> config) {
+        String cdiQualifier = null;
+        boolean qualifierRequired = false;
+        if (config != null && !config.isUnsatisfied()) {
+            final Config theConfig = config.get();
+            cdiQualifier = theConfig.getConfigValue(REACTIVE_MESSAGING_VERTX_CDI_QUALIFIER).getValue();
+            qualifierRequired = theConfig.getOptionalValue(REACTIVE_MESSAGING_VERTX_CDI_QUALIFIER_REQUIRED, Boolean.class)
+                    .orElse(Boolean.FALSE);
+        }
+        Instance<Vertx> vertxInstance = instanceOfVertx;
+        if (cdiQualifier != null && !cdiQualifier.isEmpty()) {
+            log.vertxFromCDIQualifier(cdiQualifier);
+            vertxInstance = instanceOfVertx.select(NamedLiteral.of(cdiQualifier));
```

Comment:
No Named, Identified if you want, but named has some hidden semantic you do not want here.

Reply:
IIUC, that means we need to add `io.quarkus.arc:arc` as the dependency to the subsystems in WildFly, including smallrye-reactive-messaging, smallrye-opentelemetry and the coming vertx subsystem for WildFly, because [the same qualifier](https://github.com/quarkusio/quarkus/blob/3.15.1/independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/Identified.java) needs to be used in both exposing side and consuming side IMO. I am not sure if that is what we want.

OTOH, this will be primarily used in WildFly which has a vertx instance exposed with a cdi qualifier, so probably it won't get triggered in Quarkus.

Reply:
Clement made a mistake, he meant `@Identifier`, which is present in SmallRye Common. ArC's `@Identified` is private and should not be used outside of ArC.

Reply:
Ah yes, sorry.

Reply:
Yes, let's go for the `@io.smallrye.common.annotation.Identifier`.

Reply:
Done